cmd/query/app/token_propagation_handler.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"net/http"
+	"strings"
+
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+func bearTokenPropagationHandler(logger *zap.Logger, h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		authHeaderValue := r.Header.Get("Authorization")
```

Comment:
I think we also need to be checking for `X-Forwarded-Access-Token` - as (for example) if using the OpenShift OAuthProxy - it will receive the bearer token from the UI, and if configured to do so, will [pass it in the upstream request](https://github.com/openshift/oauth-proxy#delegate-authentication-and-authorization-to-openshift-for-infrastructure) in the `X-Forwarded-Access-Token` header instead of `Authorization`.

@jpkrohling does this sound reasonable to you?

Reply:
Yes, it does. That's a common header set by reverse proxies.

Reply:
I put support for X-Forwarded-Access-Token, I tried to get token from there if Authorization header is not defined.